supervisor/core/instance_test.go

```diff
@@ -0,0 +1,103 @@
+package core
+
+import (
+	"os"
+	"os/exec"
+	"path"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+const testInstPath = "../../test_instances"
+const testInstName = "test_instance.lua"
+
+// commonStartTest run an Instance and checks its status.
+func commonStartTest(t *testing.T, isSignalIgnore bool) *Instance {
+	assert := assert.New(t)
+
+	instPath := path.Join(testInstPath, testInstName)
+	_, err := exec.LookPath(instPath)
+	assert.Nilf(err, `Unknown instance:"%v".Error: "%v"`, instPath, err)
+
+	// Create an Instance.
+	cmd := exec.Command(instPath)
+	var env []string
+	if isSignalIgnore {
+		env = append(env, "INSTSIGIGNORE=true")
+	}
+	cmd.Env = append(os.Environ(), env...)
+	inst := NewInstance(testInstName, cmd, env, false)
+
+	// Start the Instance.
+	err = inst.Start()
+	assert.Nilf(err, `Can't start an Instance.Error: "%v"`, err)
+
+	// Check status of the Instance.
+	assert.Equal(inst.Status().State, "running",
+		"The Instance is not running.")
+
+	// We need to wait for the new process to set handlers.
+	// It is necessary to update for more correct synchronization.
+	time.Sleep(100 * time.Millisecond)
+
+	return inst
+}
+
+// cleanup sends a SIGKILL signal to all test
+// Instances that remain alive after the test done.
+func cleanupInstanceTest(insts []*Instance) {
```

Comment:
`cleanupInstanceTest` -> `cleanupInstances`?

Reply:
Renamed to `cleanupTestInstances`.